lib/main.dart

```diff
@@ -1,65 +1,95 @@
 import 'package:flutter/material.dart';
+import 'package:provider/provider.dart';
 
-void main() {
-  runApp(App());
-}
+void main() => runApp(MyApp()); //runs the main application widget
+
+/// This is the main application widget.
+class MyApp extends StatelessWidget {
+  static const String _title = 'Data Offloading App';
 
-class App extends StatelessWidget {
-  // This widget is the root of your application.
   @override
   Widget build(BuildContext context) {
     return MaterialApp(
-      title: 'Data Offloading App',
+      title: _title,
       theme: ThemeData(
-        primarySwatch: Colors.green,
-        visualDensity: VisualDensity.adaptivePlatformDensity,
+        primarySwatch: Colors.green, //Theme color of the app
       ),
-      home: MyHomePage(title: 'Data Offloading App'),
+      home: MyStatefulWidget(), //"home" page
     );
   }
 }
 
-class MyHomePage extends StatefulWidget {
-  MyHomePage({Key key, this.title}) : super(key: key);
-  final String title;
+/// This is the stateful widget that the main application instantiates.
+class MyStatefulWidget extends StatefulWidget {
+  MyStatefulWidget({Key key}) : super(key: key); //muss ich noch herausfinden
 
   @override
-  _MyHomePageState createState() => _MyHomePageState();
+  _MyStatefulWidgetState createState() =>
+      _MyStatefulWidgetState(); //Because the apps basic structure is stateful and never changes its state we have to create a state.
 }
 
-class _MyHomePageState extends State<MyHomePage> {
-  int _counter = 0;
+/// This is the private State class that goes with MyStatefulWidget.
+class _MyStatefulWidgetState extends State<MyStatefulWidget> {
+  int _selectedIndex = 1;
+
+  void _onItemTapped(int index) {
+    //Function that sets the state of the bottom bar to the selected icon (index 0 for "Karte", index 1 for "Uebersicht" and index 2 for "Tasks")
 
-  void _incrementCounter() {
     setState(() {
-      _counter++;
+      _selectedIndex = index;
+      print(index);
     });
   }
 
   @override
   Widget build(BuildContext context) {
+    final List<Widget> _children = [Text("Map"), Text("Home"), Text("Task")];
     return Scaffold(
-      appBar: AppBar(
-        title: Text(widget.title),
-      ),
-      body: Center(
+      body: SafeArea(
+          child: Container(
+        padding: EdgeInsets.all(5.0), //set an edge of 5px on all sides
```

Comment:
is that needed? Try not to use pixels, instead use dynamic sizing. Or Add the padding to the settings icon